src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Controllers/AttachmentController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using TodoTemplate.Api.Contracts;
+
+namespace TodoTemplate.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    [Authorize]
+    public class AttachmentController : ControllerBase
+    {
+        private readonly AppSettings _appSettings;
+
+        public AttachmentController(IOptionsSnapshot<AppSettings> setting)
+        {
+            _appSettings = setting.Value;
+        }
+
+        [RequestFormLimits(ValueLengthLimit = int.MaxValue, MultipartBodyLengthLimit = int.MaxValue)]
+        [DisableRequestSizeLimit]
+        [HttpPost("[action]/{userName}"), AllowAnonymous]
+        public async Task<ActionResult> UploadProfilePhoto(string userName, IFormFile? file, CancellationToken cancellationToken)
+        {
+            if (file is null) throw new BadHttpRequestException("The request couldn't be processed");
+
+            await using var fileStream = file.OpenReadStream();
+            await using var memoryStream = new MemoryStream();
+            await fileStream.CopyToAsync(memoryStream, cancellationToken);
+            memoryStream.Seek(0, SeekOrigin.Begin);
+
+            if (Directory.Exists(_appSettings.UserProfilePhotoPath) is false)
+                Directory.CreateDirectory(_appSettings.UserProfilePhotoPath);
+
+            var oldPath = Path.Combine(Directory.GetFiles(_appSettings.UserProfilePhotoPath)
+                .FirstOrDefault(s => s.Contains(userName)) ?? string.Empty);
+
+            if (!string.IsNullOrEmpty(oldPath)) System.IO.File.Delete(oldPath);
+
+            var path = Path.Combine($"{_appSettings.UserProfilePhotoPath}\\{userName}.{file.FileName.Split(".").LastOrDefault()}");
+
+            await using var targetStream = System.IO.File.Create(path);
+            await memoryStream.CopyToAsync(targetStream, cancellationToken);
+
+            return Ok();
+        }
+
+        [HttpDelete("[action]/{userName}"), AllowAnonymous]
+        public ActionResult RemoveProfilePhoto(string userName, string fileName)
+        {
+            var path = Path.Combine(_appSettings.UserProfilePhotoPath, $"{userName}.{fileName.Split(".").LastOrDefault()}");
+            if (!System.IO.File.Exists(path)) return NotFound();
+
+            System.IO.File.Delete(path);
+            return Ok();
+        }
+
+        [HttpGet("[action]/{userName}"), AllowAnonymous]
+        public ActionResult GetProfilePhoto(string userName)
+        {
+            if (!Directory.Exists(_appSettings.UserProfilePhotoPath)) return Ok();
```

Comment:
Please return not found

Reply:
We may have scenarios where the user does not have a profile picture. In this case, if we return the Not Found, the program will get a server error. For this reason, this action returns an OK or return file bytes.
Do you have a new idea for this?